packages/block-editor/src/components/global-styles/background-panel.js

```diff
@@ -432,6 +432,10 @@ function BackgroundImageControls( {
 				}
 				variant="secondary"
 				onError={ onUploadError }
+				onReset={ () => {
+					closeAndFocus();
+					onResetImage();
```

Comment:
Besides this component and the Cover block, all reset callbacks are the same as `onSelect( undefined )`. I wonder if we really need a new `onReset` callback.

Reply:
We can make it optional by setting `onSelect( undefined )` as default action for `Reset` option.

Reply:
That adds more logic that `MediaReplaceFlow` needs to handle, and I was thinking the opposite; it already supports too many  "on-action" callback props.

@ramonjd, @andrewserong, what do you think?

Reply:
I'm torn.

A benefit is that consumers don't have to include their own `<MenuItem />` and `onClose()` functionality. I see that, in trunk, the menu hangs after reset

<img width="613" alt="Screenshot 2024-08-29 at 12 25 42 PM" src="https://github.com/user-attachments/assets/2f38bd5a-d56f-4d4e-aab3-a85609a79bbd">

The downside is that the "Remove" item won't be customizable for each block or file type. What if, later, the copy needs to update, e.g., `Remove [specific item]` or extra accessibility props — then we'd be loading `<MediaReplaceFlow />` again. For that reason this optimization could be premature.

Reply:
> and onClose() functionality

Child functions get passed that any way I see:

https://github.com/WordPress/gutenberg/blob/d82ce72ffc9260d3ef1693fbdc9c28ca7d7bccbc/packages/block-editor/src/components/media-replace-flow/index.js#L206-L206

Reply:
After adding "Reset" support to most media blocks in the core, I see this as more consolidation than optimization.

I'm pro-composition, but here, it seems we want to have the same feature/label across the blocks.

Reply:
Good discussion! I suppose one potential challenge with removing the `onRemove` prop and consolidating around it calling `onSelect( undefined )` is that the `MediaReplaceFlow` component appears to be a public / stable one. Would consumers be guaranteed to handle it, i.e. what if someone has a callback that always expects a value? And should consumers always have a Reset option?

I suppose one of the neat things about the `onRemove` prop is that if consumers don't provide a callback, the option will never be displayed, so it's backwards compatible in case anyone has code that wouldn't handle `onSelect( undefined )` gracefully.

But I don't feel too strongly either way on this topic 🙂

Reply:
Thanks, good point, @andrewserong!

If consumers don't expect the "Reset" button, the block editor shouldn't surprise them with one :)